py/desitarget/mock/build.py

```diff
@@ -1073,436 +1078,290 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
-                     (rzpivot, np.polyval(coeff1, rzpivot)),
-                     ((grlim[0] - 0.1 - coeff1[1]) / coeff1[0], grlim[0] - 0.1)
-                     ]
-            ax.add_patch(Polygon(verts, fill=False, ls='--', color='k'))
-        gr1 = -2.5 * np.log10( truth['FLUX_G'] / truth['FLUX_R'] )
-        rz1 = -2.5 * np.log10( truth['FLUX_R'] / truth['FLUX_Z'] )
-        gr = -2.5 * np.log10( targets['FLUX_G'] / targets['FLUX_R'] )
-        rz = -2.5 * np.log10( targets['FLUX_R'] / targets['FLUX_Z'] )
-        
-        fig, ax = plt.subplots()
-        ax.scatter(rz1, gr1, color='red', label='Noiseless Photometry')
-        ax.scatter(rz, gr, alpha=0.5, color='green', label='Noisy Photometry')
-        ax.set_xlim(-0.5, 2)
-        ax.set_ylim(-0.5, 2)
-        elg_colorbox(ax)
-        ax.legend(loc='upper left')
-        plt.show()
-        import pdb ; pdb.set_trace()
-        
-    if 'BOSS_STD' in source_data.keys():
-        boss_std = source_data['BOSS_STD'][onbrick]
-    else:
-        boss_std = None
-    select_targets_function(targets, truth, boss_std=boss_std)
-
-    keep = np.where(targets['DESI_TARGET'] != 0)[0]
-    nobj = len(keep)
-
-    if nobj == 0:
-        log.warning('No {} targets identified!'.format(target_name.upper()))
-    else:
-        log.debug('Selected {} {}s on brick {}.'.format(nobj, target_name.upper(), thisbrick))
-        targets = targets[keep]
-        truth = truth[keep]
-
-    return [targets, truth]
-
-def targets_truth_no_spectra(params, output_dir='.', realtargets=None, seed=None, verbose=False,
-                  bricksize=0.25, nproc=1, nside=16, healpixels=None):
-    """Generate a catalog of targets, spectra, and the corresponding "truth" catalog
-    (with, e.g., the true redshift) for use in simulations.
+def initialize(params, verbose=False, seed=1, output_dir="./", nproc=1, healpix_nside=16, healpixels=None):
+    """Initializes variables to prepare mock target generation.
 
     Args:
         params : dict
             Source parameters.
+        seed: int
+            Seed for the random number generatios
         output_dir : str
             Output directory (default '.').
-        realtargets : astropy.table
-            Real target catalog table, e.g. from DR3 (deprecated!).
-        bricksize : float
-            Brick size for assigning bricknames, which should match the imaging
-            team value (default 0.25 deg).
         nproc : int
             Number of parallel processes to use (default 1).
-        nside : int
+        healpix_nside : int
             Healpix resolution corresponding to healpixels (default 16).
         healpixels : numpy.ndarray or int
             Restrict the sample of mock targets analyzed to those lying inside
-            this set (array) of healpix pixels.
-
+            this set (array) of healpix pixels. Default (None)
+        
     Returns:
-        A variety of fits files are written to output_dir.
```

Comment:
What does initialize do?  "Initializes variables..." what variables?  Does it write any files anymore?  Create directories?  Change some global variables?

Reply:
I have updated all the docstrings in the new functions.